### DISCUSSION TITLE
fzf launch fuzzy find with --glob?

### DISCUSSION THREAD

**Discussion by PatschD** [LazyVim/LazyVim#5174] [General]
Hi,
currently, the default when launching the fuzzy finder is:

```lua
{ "<leader>/", LazyVim.pick("live_grep"), desc = "Grep (Root Dir)" },
```

however, in my experience, `live_grep_glob` is much more helpful.  `live_grep_glob` is like `live_grep` but with `rg --glob` support, which allows for additional filtering. eg `fzf --*.lua` searches for `fzf` only in `lua`files. 

anyway, even if it's not added as the default, i recommend adding it to the docs or something

```lua
  {
    "ibhagwan/fzf-lua",
    keys = {
      { "<leader>/", LazyVim.pick("live_grep_glob"), desc = "Grep Glob With Pattern(Root Dir)" },
    },
  },
```

**Comment by skoch13** [comment]
One can add the following, so `live_grep` acts as `--glob` when `--` is specified, e.g. `hello -- -tts` will search for hello strings in `ts` files

```
  {
    "ibhagwan/fzf-lua",
    opts = {
      grep = {
        rg_glob = true,
        -- first returned string is the new search query
        -- second returned string are (optional) additional rg flags
        -- @return string, string?
        rg_glob_fn = function(query, opts)
          local regex, flags = query:match("^(.-)%s%-%-(.*)$")
          -- If no separator is detected will return the original query
          return (regex or query), flags
        end,
      },
    },
  },
```

**Comment by ibhagwan** [comment]
@PatschD, add `grep = { rg_glob = true }` to your opts and all grep pickers will have globs enabled, `rg_glob_fn` is only if you want to change the default glob parser (to pass other flags).
